include/pybind11/pybind11.h

```diff
@@ -975,27 +975,27 @@ class module_ : public object {
             /* m_clear */    nullptr,
             /* m_free */     nullptr
         };
-        m_ptr = PyModule_Create(def);
-        if (m_ptr == nullptr)
-            pybind11_fail("Internal error in module_::module_()");
-        inc_ref();
-    }
+        auto m = PyModule_Create(def);
+        if (m == nullptr)
+            pybind11_fail("Internal error in module_::create_extension_module()");
+        // TODO: Should be reinterpret_steal, but Python also steals it again when returned from PyInit_...
+        return reinterpret_borrow<module_>(m);
+#else
+        // Ignore module_def *def; only necessary for Python 3
+        (void) def;
+        auto m = Py_InitModule3(name, nullptr, options::show_user_defined_docstrings() ? doc : nullptr);
+        if (m == nullptr)
+            pybind11_fail("Internal error in module_::create_extension_module()");
+        return reinterpret_borrow<module_>(m);
 #endif
```

Comment:
I thought about that, but decided not to because of the `// TODO: Should be reinterpret_steal, but Python also steals it again when returned from PyInit_...` comment (which only holds for Python 3; in Python 2, it should be `reinterpret_borrow`).

Reply:
> Did you see my question about `pybind11_fail` vs `throw error_already_set()` from yesterday?

Yes. Wait, let me reply in that thread.